searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp

```diff
@@ -0,0 +1,623 @@
+        set_v8_geo_positions(false);
+        expect_insert(R"({"y":750000,"x":500000})", position);
+        set_v8_geo_positions(true);
+    }
+    {
+        SCOPED_TRACE("partial position");
+        StructFieldValue position(get_data_type("position"));
+        position.setValue("x", IntFieldValue(500000));
+        expect_insert(R"({"x":500000})", position);
+    }
+    {
+        SCOPED_TRACE("empty position");
+        StructFieldValue position(get_data_type("position"));
+        expect_insert("{}", position);
+    }
+}
+
+TEST_F(SlimeFillerTest, insert_uri)
+{
+    StructFieldValue uri(get_data_type("url"));
+    uri.setValue("all", StringFieldValue("http://www.example.com:42/foobar?q#frag"));
+    uri.setValue("scheme", StringFieldValue("http"));
+    uri.setValue("host", StringFieldValue("www.example.com"));
+    uri.setValue("port", StringFieldValue("42"));
+    uri.setValue("path", StringFieldValue("foobar"));
+    uri.setValue("query", StringFieldValue("q"));
+    uri.setValue("fragment", StringFieldValue("frag"));
+    expect_insert(R"("http://www.example.com:42/foobar?q#frag")", uri);
+}
+
+TEST_F(SlimeFillerTest, insert_predicate)
+{
+    auto input = std::make_unique<Slime>();
+    Cursor &obj = input->setObject();
+    obj.setLong(Predicate::NODE_TYPE, Predicate::TYPE_FEATURE_SET);
+    obj.setString(Predicate::KEY, "foo");
+    Cursor &arr = obj.setArray(Predicate::SET);
+    arr.addString("bar");
+    PredicateFieldValue value(std::move(input));
+    expect_insert(R"("'foo' in ['bar']\n")", value);
+}
+
+TEST_F(SlimeFillerTest, insert_tensor)
+{
+    TensorDataType   data_type(ValueType::from_spec("tensor(x{},y{})"));
+    TensorFieldValue value(data_type);
+    value = make_tensor(TensorSpec("tensor(x{},y{})").add({{"x", "4"}, {"y", "5"}}, 7));
+    expect_insert(make_slime_tensor_string(*value.getAsTensorPtr()), value);
+    expect_insert(R"("0x")", TensorFieldValue());
+}
+
+TEST_F(SlimeFillerTest, insert_reference)
+{
+    {
+        SCOPED_TRACE("normal reference");
+        ReferenceFieldValue value(get_as_ref_type("Reference<target_dummy_document>"),
+                                  DocumentId("id:ns:target_dummy_document::foo"));
+        expect_insert(R"("id:ns:target_dummy_document::foo")", value);
+    }
+    {
+        SCOPED_TRACE("empty reference");
+        ReferenceFieldValue value(get_as_ref_type("Reference<target_dummy_document>"));
+        expect_insert(R"("")", value);
+    }
+}
+
+TEST_F(SlimeFillerTest, insert_array)
+{
+    auto array = make_array();
+    expect_insert(R"(["foo","bar","baz"])", array);
+}
+
+TEST_F(SlimeFillerTest, insert_array_filtered)
+{
+    auto array = make_array();
+    expect_insert_filtered(R"(["foo","bar","baz"])", array, {0, 1, 2});
+    expect_insert_filtered("null", array, {});
+    expect_insert_filtered(R"(["foo"])", array, {0});
+    expect_insert_filtered(R"(["bar"])", array, {1});
+    expect_insert_filtered(R"(["baz"])", array, {2});
```

Comment:
Consider adding a single test case for 2 elements as well.

Reply:
Done.